pkg/integrations/v2/mssql/sql_exporter.go

```diff
@@ -0,0 +1,149 @@
+package snowflake_exporter
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/burningalchemist/sql_exporter"
+	"github.com/burningalchemist/sql_exporter/config"
+	"github.com/go-kit/log"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/prometheus/common/model"
+
+	mssql_common "github.com/grafana/agent/pkg/integrations/mssql/common"
+	integrations_v2 "github.com/grafana/agent/pkg/integrations/v2"
+	"github.com/grafana/agent/pkg/integrations/v2/common"
+	"github.com/grafana/agent/pkg/integrations/v2/metricsutils"
+)
+
+// DefaultConfig is the default config for the mssql v2 integration
+var DefaultConfig = Config{
+	MaxIdleConnections: 3,
+	MaxOpenConnections: 3,
+	Timeout:            10 * time.Second,
+}
+
+// Config is the configuration for the mssql v2 integration
+type Config struct {
+	ConnectionString   string               `yaml:"connection_string,omitempty"`
+	MaxIdleConnections int                  `yaml:"max_idle_connections,omitempty"`
+	MaxOpenConnections int                  `yaml:"max_open_connections,omitempty"`
+	Timeout            time.Duration        `yaml:"timeout,omitempty"`
+	Common             common.MetricsConfig `yaml:",inline"`
+}
+
+func (c Config) validate() error {
```

Comment:
Could be possible somehow reuse this function instead of duplicating?

Reply:
I pulled the shared config options into a shared config struct to dedup.